Settings.StyleCop

```diff
@@ -27,6 +40,18 @@
             <BooleanProperty Name="Enabled">False</BooleanProperty>
           </RuleSettings>
         </Rule>
+        <Rule Name="UseBuiltInTypeAlias">
+          <!-- it's OK to use 'String' for API calls -->
+          <RuleSettings>
+            <BooleanProperty Name="Enabled">False</BooleanProperty>
+          </RuleSettings>
+        </Rule>
+        <Rule Name="CodeMustNotContainMultipleStatementsOnOneLine">
```

Comment:
(!) We never use multiple statements on one line.

Reply:
I see one pattern in code base that violates this rule:
```
switch (...)
{
   case a: <very-simple-statement>; break;
   case b: <very-simple-statement>; break;
   ...
   default: break;
}
```
I personally think this is fine to have, but open to more comments.

Reply:
I think it's most often clearer to break them into separate lines... although I do wish C# had a less cluttered way to do one-line cases. :/

Reply:
Prefer single statements per line myself, but cleaning up existing code for this one could be too painful.

Reply:
I'd prefer to have one pattern for simple and complex code. I mean multiline. Because later we can add a fix with multiline code - in the case we will be forced to reformat the whole code block. It mixes functional and style changes that complicates a code review.

Reply:
OK, I will keep this rule enabled.

Reply:
Is there any way of doing targeted opt-out?

The most compelling example is single returns in a switch - something that will only be more common/used with C# 8 and the switch expression.
```csharp
switch (...)
{
   case a: return 1;
   case b: return 2;
   ...
   default: return 0;
}
```
I find that easier to read (more like a table) then when the statements are on different lines.

and C#8 will bring
```csharp
return obj switch
{
  string s => $"A string: {s}",
  int i    => $"An integer {I}",
  _        => $"something else" 
};
```
I think it would be a shame to not allow this where it makes sense.
The same thing with using spaces like in the example above to improve clarity when there are tables.

Maybe just use attributes to opt out?

Reply:
Such attributes don't add readability. 
CodeFactor is not mandatory and it is not problem if it sometimes return false positives.